### PR TITLE
Add test script for States Func and fix missing env variable

### PR DESCRIPTION
**Pull request by bgfoshay**
## What’s changing?

Fixed missing environment variable.

Also adding a bash script that can be used by admin for inserting/updating a StateInfo record. Ryan and I noted during the last deployment that until we provide an admin page in the web app, Tatiana or whoever is deploying, needs a means to call the API to insert/update records and make sure email & phone get encrypted in StateInfo table. 

## Why?

Closes [Nac-1916](https://e-jira.fns.usda.net/jira-prod/browse/NAC-1916)

## This PR has:

- [x] **JIRA ticket(s)** (_Must be listed in PR request and branch name should follow convention NAC-123-description_)<description>

- [x] **Security Related Changes** (_Security Tab questions have been answered on JIRA ticket(s)_)

- [x] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [x] **Documented API changes** (_required on public classes and methods, and elsewhere as appropriate_)
    - [ ] External API documentation - OpenApi docs and Generated Docs (e.g. https://github.com/18F/piipan/tree/dev/docs/openapi) 
    - [ ] Internal API documentation (e.g. https://github.com/18F/piipan/tree/dev/metrics/docs/openapi)   

- [x] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [x] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [x] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)
    - [x] Have IaC scripts run successfully against an existing deployed environment?
    - [x] Have IaC scripts run successfully against a newly deployed environment?
    - [x] Have any new bash scripts been given the appropriate permissions? (e.g. git update-index --chmod=+x $script)
    - [ ] Do the IaC changes require deleting existing resources?

## Anything else?

_Add other details that are helpful for review here_

## Code Review Checklist:

- [x] Automated checks all pass? (code builds successfully, tests pass, test coverage meets/exceeds threshold, etc.) 
- [x] Have all requirements and Acceptance Criteria been met?
- **If there are code changes**
    - [x] Is there no unnecessary complexity? Code is readable, maintainable, etc.
    - [x] Do the changes follow SOLID principles?
    - [x] Are the changes robust? Concurrency considered? Performance? Security (e.g. no sql injections)?
    - [x] Are newly added dependencies needed and appropriate? Are dependency files (e.g. packages.json & packages.lock.json) updated?
    - [ ] For any new projects/solutions, have they been added to code coverage files?
    - [ ] Public classes & methods are documented    
- **If there are API changes**
    - [ ] Does API make sense? Useful and concise?
    - [ ] Are existing API interfaces/contracts intact? i.e. no breaking changes
    - [ ] Are API tests included?
    - [ ] Are API responses appropriate and documented? (200s, 201, 404s, etc.)
    - [ ] Are internals kept private and not leaked into the API?
    - [ ] External API documentation - OpenApi docs and Generated Docs 
    - [ ] Internal API documentation    
- **If there are IAC changes**
    - [x] IAC Scripts are readable & maintainable
    - [x] IAC Scripts are documented
    - [x] IAC scripts pass required checks
    - [x] IAC scripts were run successfully by author
- **General Logging Questions**
    - [x] Do new capabilities (or changed capabilities) include logging?
    - [x] Is PII not logged?
- **General Error Handling Questions**
    - [x] Has exception handling been added for new capabilities (or changed capabilities)?
    - [ ] Are errors/exceptions properly communicated to end-users? Were there requirements for user facing error messages?
    - [x] Are errors/exceptions logged?
    - [x] Is PII kept out of error/exception logging?
- **If there are Developer Documentation Changes**
    - [ ] General Project documentation (under piipan\docs. e.g. iac.md, architecture.md, etc.)
    - [ ] Github documentation additions/changes (Readmes)
- **General automated test questions**
    - [ ] End-to-End Tests provided (if code implements new user story)
    - [x] Are tests sufficient?
    - [x] Corner cases considered?
    - [ ] Are performance tests needed?

### CODE REVIEW

File: states/src/Piipan.States/Piipan.States.Core/DataAccessObjects/StateInfoDao.cs [hunk 1]
```diff
@@ -31,15 +32,17 @@ public async Task<IState> GetStateById(string id)
         {
             try
             {
-                return await _databaseManager.PerformQuery(async connection =>
-                {
-                    return await connection
-                        .QuerySingleAsync<StateInfoDbo>(@"
+                StateInfoDbo result = await _databaseManager.PerformQuery(async connection =>
+                                   {
```
Comment (ccaprio97) [inline states/src/Piipan.States/Piipan.States.Core/DataAccessObjects/StateInfoDao.cs:36]: nitpick: is this spacing on purpose?

File: states/src/Piipan.States/Piipan.States.Core/DataAccessObjects/StateInfoDao.cs [hunk 2]
```diff
@@ -56,15 +59,18 @@ public async Task<IState> GetStateByName(string state)
...
+                                       return await connection
```
Comment (ccaprio97) [inline states/src/Piipan.States/Piipan.States.Core/DataAccessObjects/StateInfoDao.cs:64]: nitpick: same as previous comment

File: states/src/Piipan.States/Piipan.States.Core/DataAccessObjects/StateInfoDao.cs [hunk 3]
```diff
@@ -87,15 +93,29 @@ public async Task<IEnumerable<IState>> GetStates()
...
+        {
+            try
+            {
+                state.Email = _cryptographyClient.DecryptFromBase64String(state.Email);
+                state.Phone = _cryptographyClient.DecryptFromBase64String(state.Phone);
+                state.EmailCc = _cryptographyClient.DecryptFromBase64String(state.EmailCc);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogWarning(ex, $"Failed to decrypt values for {state.State}.");
```
Comment (ccaprio97) [inline states/src/Piipan.States/Piipan.States.Core/DataAccessObjects/StateInfoDao.cs:115]: is it worth it to add which part broke it? email vs phone vs emailCc?

File: states/tests/Piipan.States.Core.Integration.Tests/StateInfoDaoTests.cs [hunk 4]
```diff
@@ -50,17 +39,70 @@ public async void GetStateByNameTest()
                 InsertState();
 
                 var expected = GetLastStateName();
+                var storedEmail = GetLastStateEmail();
```
Comment (dborgen2) [inline states/tests/Piipan.States.Core.Integration.Tests/StateInfoDaoTests.cs:42]: Nitpick/Question: I know this is the pattern we've been using, but I've never been too keen on having functions that call into the database that are only used by tests rather than using our actual Dao. In this particular example, we see 3 calls to functions that our actual production code doesn't use (InsertState, GetLastStateName, GetLastStateEmail), when I think we could just create a StateInfoDto, call UpsertState, and check the StateInfoDto's email with result.Email at the end.

Not sure if it's worth changing here, but a discussion that we may want to have more broadly as a dev team, as I have seen this pattern a lot while working with integration tests lately.

File: states/tests/Piipan.States.Core.Integration.Tests/DbFixture.cs [hunk 5]
```diff
@@ -43,6 +43,19 @@ public void InsertState()
             }
         }
 
+        public void InsertStateWithInvalidEncryptionValues()
```
Comment (dborgen2) [inline states/tests/Piipan.States.Core.Integration.Tests/DbFixture.cs:46]: Nitpick/Question: Similar comment that I had earlier about just using UpsertState instead of creating these test methods. Do we think it could be useful here?

**Review by dborgen2 — APPROVED** [review]
👍 It looks good to me. Some broader discussions about the best way to test integration tests I think should be had, but this works and is consistent with how we usually do it.

File: states/tests/Piipan.States.Core.Integration.Tests/StateInfoDaoTests.cs (hunk 4 above)
Comment (bgfoshay) [reply states/tests/Piipan.States.Core.Integration.Tests/StateInfoDaoTests.cs:42]: I completely agree. This pattern is messy. Part of the issue with this is that the decryption is being handled at the Dao level and the encryption is handled at the top level Func code. Rather than in a service level in-between. We have a ticket for this work. https://e-jira.fns.usda.net/jira-prod/browse/NAC-1814. I think we should incorporate your recommended changes as part of that work. I'll add a reference to your comments in the ticket.
